src/test/java/org/jenkinsci/plugins/workflow/steps/GeneralNonBlockingStepExecutionTest.java

```diff
@@ -115,6 +116,21 @@
         assertThat(logging.getRecords(), empty());
     }
 
+    @Issue("JENKINS-58878")
+    @Test public void shouldNotHang() throws Exception {
+        int iterations = 50;
+        startExit = new Semaphore(iterations); // Prevents the semaphores from blocking inside of the slowBlock step.
+        endExit = new Semaphore(iterations);
```

Comment:
```suggestion
        endExit.release(iterations);
```

perhaps?

Reply:
Good point, I always get confused and think that the permits are limited at construction time. Fixed in 707eaaf.